string fire_guard
string transition
string token
---
bool guard
